src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py

```diff
@@ -4209,5 +4209,19 @@ def test_vmss_update_automatic_repairs_with_health_extension(self, resource_grou
                  ])
 
 
+class VMCreateNSGRule(ScenarioTest):
+
+    @ResourceGroupPreparer(name_prefix='cli_test_vm_create_nsg_rule_')
+    def test_vm_create_nsg_rule(self, resource_group):
+        self.kwargs.update({
+            'vm': 'vm1'
+        })
+
+        self.cmd('vm create -g {rg} -n {vm} --image centos --nsg-rule NONE')
+        self.cmd('network nsg show -g {rg} -n {vm}NSG', checks=[
+            self.not_exists('securityRules')
```

Comment:
In my test `securityRules` is an empty list, instead of not exist or `None`.

```
az network nsg show --ids /subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590/resourceGroups/{}/providers/Microsoft.Network/networkSecurityGroups/{}
  "securityRules": [],
```